test/e2e/retry.go

```diff
@@ -51,5 +45,5 @@ func retryWithExponentialBackOff(fn func() error) error {
 			return false, err
 		}
 	}
-	return wait.ExponentialBackoff(backoff, retryFn(fn))
```

Comment:
@jackfrancis do we need to update the retryFn so it doesn't return an err according to https://github.com/kubernetes-sigs/cluster-api/pull/6437/files?

Reply:
Yep. https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/2261

Thanks for tracking!